x/feegrant/keeper/grpc_query.go

```diff
@@ -109,19 +99,31 @@ func (q Keeper) AllowancesByGranter(c context.Context, req *feegrant.QueryAllowa
 
 	ctx := sdk.UnwrapSDKContext(c)
 
-	store := q.storeService.OpenKVStore(ctx)
-	prefixStore := prefix.NewStore(runtime.KVStoreAdapter(store), feegrant.FeeAllowanceKeyPrefix)
-	grants, pageRes, err := query.GenericFilteredPaginate(q.cdc, prefixStore, req.Pagination, func(key []byte, grant *feegrant.Grant) (*feegrant.Grant, error) {
-		// ParseAddressesFromFeeAllowanceKey expects the full key including the prefix.
-		granter, _ := feegrant.ParseAddressesFromFeeAllowanceKey(append(feegrant.FeeAllowanceKeyPrefix, key...))
-		if !bytes.Equal(granter, granterAddr) {
-			return nil, nil
-		}
-
-		return grant, nil
-	}, func() *feegrant.Grant {
-		return &feegrant.Grant{}
-	})
+	var grants []*feegrant.Grant
+	_, pageRes, err := query.CollectionFilteredPaginate(ctx, q.FeeAllowance, req.Pagination,
+		func(key collections.Pair[sdk.AccAddress, sdk.AccAddress], grant feegrant.Grant) (include bool, err error) {
+			if !sdk.AccAddress(granterAddr).Equals(key.K2()) {
+				return false, nil
+			}
+
+			grants = append(grants, &grant)
+			return false, nil
+		},
+	)
```

Comment:
@testinginprod how to migrate the filteredPagination to collections? I did the above but it is counting all the records from state instead which matched the condition specified.

Reply:
I'm not understanding the error? could you make an example?

Reply:
Assume these are the five grants exist in state
```
[
	{
		"grantee": "granteeAddr1",
		"granter": "granterAddr1"
	},
	{
		"grantee": "granteeAddr2",
		"granter": "granterAddr1"
	},
	{
		"grantee": "granteeAddr3",
		"granter": "granterAddr1"
	},
	{
		"grantee": "granteeAddr4",
		"granter": "granterAddr1"
	},
	{
		"grantee": "granteeAddr5",
		"granter": "granterAddr2"
	},
]
```
I want to query the grants by granter address (where the key structure is `key collections.Pair[grantee, granter]`), the possible solution is to iterate all grants and match the key.Key2() (which is granter address) with the given input.

I am doing the same with the above code, but the problem is the total count in the pagination response is 5 instead of 1 when I query it with `granterAddr2` (`granterAddr2` has only 1 grant exist in the state)

Reply:
@atheeshp this is, as far i understand, the expect behaviour in pagination

Reply:
hmm @testinginprod, If I want the count total of only matched records, what changes need to be done?

Reply:
earlier to this, `GenericFilteredPaginate` used to work as I mentioned.